### PR TITLE
Bump graphql-java to 16.2, do not include external dependencies #564 #589

### LINKED ISSUE #589: graphql-java version not compatible with graphql-spring-boot-starter
I upgraded my implementation project to the latest graphql-spring-boot-starter version but this caused a bean creation error when used with the graphql generated code because of conflicting versions of graphql-java code generator. Would it be possible to upgrade the version of graphql-java in graphql-java-codegen to 16.2?

code-generation project:
```
<dependency>
    <groupId>io.github.kobylynskyi</groupId>
    <artifactId>graphql-java-codegen</artifactId>
    <version>4.1.0</version>
</dependency>
```
implementation project:
 ```
<dependency>
    <groupId>com.graphql-java-kickstart</groupId>
    <artifactId>graphql-spring-boot-starter</artifactId>
    <version>11.0.0</version>
</dependency>
```

error:
```
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'graphQLSchemaProvider' defined in class path resource [graphql/kickstart/spring/web/boot/GraphQLWebAutoConfiguration.class]: Unsatisfied dependency expressed through method 'graphQLSchemaProvider' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'graphQLSchema' defined in class path resource [graphql/kickstart/tools/boot/GraphQLJavaToolsAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [graphql.schema.GraphQLSchema]: Factory method 'graphQLSchema' threw exception; nested exception is java.lang.NoClassDefFoundError: graphql/collect/ImmutableKit
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'graphQLSchema' defined in class path resource [graphql/kickstart/tools/boot/GraphQLJavaToolsAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [graphql.schema.GraphQLSchema]: Factory method 'graphQLSchema' threw exception; nested exception is java.lang.NoClassDefFoundError: graphql/collect/ImmutableKit
```
The error was solved when I explicitly managed the dependency of graphql-java to 16.2.

```
<dependency>
    <groupId>com.graphql-java</groupId>
    <artifactId>graphql-java</artifactId>
    <version>16.2</version>
</dependency>
```

### PR DESCRIPTION
**Pull request by kobylynskyi**
Closes #589

### CODE REVIEW

**Comment by kobylynskyi** [comment]
@jxnu-liguobin could you please take a look at the failed sbt-test step? Thank you.

**Comment by kobylynskyi** [comment]
@jxnu-liguobin could you please take a look at the failed sbt-test step? Thank you.

**Comment by jxnu-liguobin** [comment]
@kobylynskyi I can't find the problem. I'll pull it down and have a look

**Comment by jxnu-liguobin** [comment]
My guess is that all the dependencies have been changed to compile time, so the SBT plug-in should automatically import these dependencies for users, but not for the plug-in itself. Instead, it should modify `GraphQLCodegenPluginDependencies`

I see that in Maven and Gradle, you add these packages manually, but unfortunately, the scala sample project does not.

File: plugins/sbt/graphql-java-codegen-sbt-plugin/build.sbt [hunk 1]
```diff
@@ -31,7 +31,11 @@ lazy val `graphql-codegen-sbt-plugin` = Project(id = "graphql-codegen-sbt-plugin
       pushChanges
     ),
     libraryDependencies ++= Seq(
-      "io.github.kobylynskyi" % "graphql-java-codegen" % (version in ThisBuild).value
+      "io.github.kobylynskyi" % "graphql-java-codegen" % (version in ThisBuild).value,
+      "org.freemarker" % "freemarker" % "2.3.31",
```
Comment (jxnu-liguobin) [inline plugins/sbt/graphql-java-codegen-sbt-plugin/build.sbt:35]: remove

Comment (jxnu-liguobin) [reply plugins/sbt/graphql-java-codegen-sbt-plugin/build.sbt:35]: I'll test it first. Later

**Comment by jxnu-liguobin** [comment]
The problem was found. That example has used graphql-java, but did not import the dependency. (there is no such dependency in codgen).

add it here(example-client)
image

**Comment by kobylynskyi** [comment]
Thanks @jxnu-liguobin

**Pull request merged**